ui-admin/src/providers/AdminUserProvider.tsx

```diff
@@ -26,14 +27,24 @@ export const useAdminUserContext = () => {
  */
 export default function AdminUserProvider({ portalShortcode, children }:
 {portalShortcode: string, children: React.ReactNode}) {
+  const { user } = useUser()
   const [userState, setUserState] = useState<AdminUserContextT>({ users: [], isLoading: true })
   useEffect(() => {
-    Api.fetchAdminUsersByPortal(portalShortcode).then(result => {
-      setUserState({
-        users: result,
-        isLoading: true
+    if (user?.superuser) {
+      Api.fetchAdminUsers().then(result => {
+        setUserState({
+          users: result,
+          isLoading: true
+        })
```

Comment:
What happens if a superuser not in your portal makes a change that shows up in the audit table? Their username wouldn't show up right? Maybe in the audit table we could have the text default to `Juniper Admin` or something like that if it doesn't show up

Reply:
If I'm understanding the question correctly, this shouldn't be an issue because superusers have access to all portals. There's a related case where if a non-superuser views this table, they won't have access to load the list of superusers, so those entries will display the text "superuser (<first few digits of user id>)"